packages/language-server/src/plugins/css/CSSClassDefinitionLocator.ts

```diff
@@ -0,0 +1,167 @@
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Class Directive format:
+     * class:active="{current === 'foo'}"
+     */
+    public IsDirectiveFormat() {
+        if (this.initialNodeAt?.type == 'Class' && this.initialNodeAt?.parent?.type == 'Element') {
+            return true;
+        }
+
+        return false;
+    }
+
+    public GetStandardFormatClassName() {
+        const testEndTagRange = Range.create(
+            Position.create(this.position.line, 0),
+            Position.create(this.position.line, this.position.character)
+        );
+        const text = this.document.getText(testEndTagRange);
+
+        let loopLength = text.length;
+        let testPosition = this.position.character;
+        let spaceCount = 0;
+
+        //Go backwards until hitting a " and keep track of how many spaces happened along the way
+        while (loopLength) {
+            const testEndTagRange = Range.create(
+                Position.create(this.position.line, testPosition - 1),
+                Position.create(this.position.line, testPosition)
+            );
+            const text = this.document.getText(testEndTagRange);
+            if (text === ' ') {
+                spaceCount++;
+            }
+
+            if (text === '"') {
+                break;
+            }
+
+            testPosition--;
+            loopLength--;
+        }
+
+        const cssClassName = this.initialNodeAt?.data.split(' ')[spaceCount];
+
+        return this.GetDefinitionRangeWithinStyleSection(`.${cssClassName}`);
+    }
+
+    public GetDefinitionRangeWithinStyleSection(targetClass: string) {
+        let indexOccurence = this.document.content.indexOf(targetClass, 0);
+
+        while (indexOccurence >= 0) {
+            if (this.IsOffsetWithinStyleSection(indexOccurence)) {
+                const startPosition = this.document.positionAt(indexOccurence);
+                const targetRange = Range.create(
+                    startPosition,
+                    Position.create(
+                        startPosition.line,
+                        startPosition.character + targetClass.length
+                    )
+                );
+                indexOccurence = this.document.content.indexOf(targetClass, indexOccurence + 1);
+
+                if (!this.IsExactClassMatch(targetRange)) {
+                    continue;
+                }
+
+                return targetRange;
+            }
+        }
+    }
+
+    public IsOffsetWithinStyleSection(offsetPosition: number) {
+        if (this.document.styleInfo) {
+            if (
+                offsetPosition > this.document.styleInfo?.start &&
+                offsetPosition < this.document.styleInfo?.end
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public IsExactClassMatch(testRange: Range) {
+        //Check nothing before the test position
+        if (testRange.start.character > 0) {
```

Comment:
This would fail for something like `.foo.bar { .. }` or `.foo .bar { .. }`. In general it's a little brittle to only use text matching here. The vscode-css-language-service which is used in the CSSPlugin provides an AST I think, but it's private. This puts us at risk of breaking changes but I feel it's more robust to use that one instead and fall back to this text-based matching only if parsing fails (because a different preprocessor is used for example).

Reply:
I had a feeling about that one. The case I thought would be the most straightforward turned out, not. I will try to improve it. I was looking for a way to pull the text for an entire line but couldn't find it

Reply:
Note that svelte doesn't recognize the .foo .bar case either

![image](https://user-images.githubusercontent.com/5598169/173704164-5eac6463-2a25-4a72-8e5d-fd13f85410f8.png)

Reply:
@dummdidumm So this actually did already work for **.foo .bar** but did not for **.foo.bar**

Try the latest, it handles all sorts of combinations. I feel this is a good compromise especially since I really don't feel that style's inside a svelte component would/should be very complex. If after testing you still feel like we need more horsepower I can look into the CSS AST(just point me in the right direction for that)